engine/src/main/java/com/datatorrent/stram/plan/logical/LogicalPlanConfiguration.java

```diff
@@ -1023,15 +1774,30 @@ private StramElement getElement(String value, Conf conf) {
     return element;
   }
 
-  private String getCompleteKey(String[] keys, int start) {
+  /**
+   * This constructs a string from the keys in the given keys array starting from
+   * the start index inclusive until the end of the array.
+   * @param keys The keys from which to construct a string.
+   * @param start The token to start creating a string from.
+   * @return The completed key.
+   */
+  private static String getCompleteKey(String[] keys, int start) {
     return getCompleteKey(keys, start, keys.length);
   }
 
-  private String getCompleteKey(String[] keys, int start, int end) {
+  /**
+   * This constructs a string from the keys in the given keys array starting from
+   * the start index inclusive until the specified end index exclusive.
+   * @param keys The keys from which to construct a string.
+   * @param start The token to start creating a string from.
+   * @param end 1 + the last index to include in the concatenation.
+   * @return The completed key.
+   */
+  private static String getCompleteKey(String[] keys, int start, int end) {
     StringBuilder sb = new StringBuilder(1024);
```

Comment:
We know exactly how many elements we need, why are we allocating 1024 instead of (end-start) ?